sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/RequestResponseChannel.java

```diff
@@ -366,42 +387,45 @@ private void settleMessage(Message message) {
     }
 
     private void handleError(Throwable error, String message) {
-        if (hasError.getAndSet(true)) {
+        if (this.hasError.getAndSet(true)) {
             return;
         }
 
         logger.error("connectionId[{}] linkName[{}] {} Disposing unconfirmed sends.", connectionId, linkName, message,
             error);
 
-        endpointStates.emitError(error, (signalType, emitResult) -> {
+        this.endpointStates.emitError(error, (signalType, emitResult) -> {
             logger.warning("connectionId[{}] linkName[{}] signal[{}] result[{}] Could not emit error to sink.",
                 connectionId, linkName, signalType, emitResult);
             return false;
         });
 
-        unconfirmedSends.forEach((key, value) -> value.error(error));
-        unconfirmedSends.clear();
+        terminateUnconfirmedSends(error);
 
         closeAsync().subscribe();
     }
 
     private void onTerminalState(String handlerName) {
-        if (pendingDisposes.get() == 0) {
+        if (this.pendingLinkTerminations.get() == 0) {
```

Comment:
I wonder if it is possible for someone else to be performing decrementAndGet before we get to the next line since only this `get()` operation is atomic. maybe `if (this.pendingLinkTerminations.get() <= 0)`?

Reply:
Good point, will switch to `<= 0`